### PR TITLE
add public cloudfront behaviour

### PR DESCRIPTION
**Pull request by skycremaschinin**
Solves #207

### CODE REVIEW

**Comment by danielcondemarin** [comment]
@skycremaschinin Thank you for the PR, but I don't think this will work 🤔 

The new path pattern you've added means requests to `/public/*` will be proxied to S3 but what we need is requests to `/*` that do not match any pages or apis are proxied to S3 public folder. 

To get this working I believe we need 2 things:
1 - Add to the build manifest all files inside `public/` at any level deep. Currently it uses `readdir` which only returns the files directly under the folder. See https://github.com/danielcondemarin/serverless-next.js/blob/0a03156ecafddab71716e3df8d573a12a794d3e7/packages/serverless-nextjs-component/serverless.js#L30

2 - Copy any files inside `public/static` over to `static` folder to keep backwards compatibility. 

@jayarnielsen Since you raised the original issue how does the above sound? Hopefully I didn't miss anything.

**Comment by skycremaschinin** [comment]
@danielcondemarin i was missing the real meaning of public folder:

> Any files in the public directory will be mapped to the root of the domain.

because when public folder is found in src folder a public folder is created on s3 bucket.

Why not simply copy the local public folder to bucket root level?
this would also cover:

> Because public also covers the static directory use case we have decided to deprecate the static directory in favor of creating a public/static folder with the same functionality.

static folder would be copied to s3 root level, and static/* route would work as expected.

**Comment by danielcondemarin** [comment]
> Why not simply copy the local public folder to bucket root level?

That's what I meant by adding to the build manifest all files inside public/ any level deep. i.e. copy the entire folder including sub dirs etc. The Lambda@Edge function then takes care of checking whether there is a file in the public folder for the request path https://github.com/danielcondemarin/serverless-next.js/blob/master/packages/serverless-nextjs-component/default-lambda-handler.js#L45

Basically just ensuring the entire folder along with subdirs etc. is copied over should be enough to get this working. 
The reason why is implemented as it is currently is because originally the public folder was meant only for top level file resources like `/robots` `/manifest` etc. It didn't need to support nested routes like images/*

**Comment by skycremaschinin** [comment]
ok, got it.
i'll do some tests and open a new pr.
